middleware/dnstap/msg/msg.go

```diff
@@ -0,0 +1,89 @@
+// Package msg helps to build a dnstap Message.
+package msg
+
+import (
+	"errors"
+	"net"
+	"time"
+
+	"github.com/coredns/coredns/request"
+
+	tap "github.com/dnstap/golang-dnstap"
+	"github.com/miekg/dns"
+)
+
+// Data helps to build a dnstap Message.
+// It can be transformed into the actual Message using this package.
+type Data struct {
```

Comment:
Yes, in dnstap command-line tool's code, the combinaison of fields displayed for each event type falls under the fields described in `msg.Data`. And it is handy because [tap.Message](https://godoc.org/github.com/dnstap/golang-dnstap#Message) fieds are only pointers.

Argeed for the methods.